coordinator/impl/k8s_client.go

```diff
@@ -85,16 +91,23 @@ type clientImpl[Resource resource] struct {
 	clientFunc func(string) ResourceInterface[Resource]
 }
 
-func (c *clientImpl[Resource]) Upsert(namespace string, resource *Resource) (result *Resource, err error) {
+func (c *clientImpl[Resource]) Upsert(namespace, name string, resource *Resource) (*Resource, error) {
```

Comment:
Sorry, I am considering this name. upsert means update and insert. :/ is it okay to use a patch inside this function?

Reply:
I think it's ok. patch is one kind of update, right?